### ISSUE TITLE
Wrong result for adding two arrays

### ISSUE THREAD

**Issue by mihai-varga** [weld-project/weld#315]
The code below produces the wrong result when the size of the input vectors is at least 16385. The code adds two vectors and prints the results. When size >= 16385 there will be, on my system, size % 4 wrong values at the end of the result vector. So for 16385 I get one wrong value of 0.
Full C code: https://pastebin.com/Uz62bD0d , and the weld snippet: 
```
|x:vec[i64], y:vec[i64]|
let add = result(
   for (zip(x, y), appender[i64], |b, i, n|
	   merge(b, n.$0 + n.$1)
	)
);
{add}
```

Weld: latest version, tried with both debug and release
Linux: fedora 4.15.3-200.fc26.x86_64
Clang: tried with both 4.0.1 and 5.0.1
CPU: Intel(R) Core(TM) i5-7260U CPU @ 2.20GHz
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb invpcid_single pti tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid mpx rdseed adx smap clflushopt intel_pt xsaveopt xsavec xgetbv1 xsaves dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp

I'm not quite sure if the issue is with Weld or my system. Are you able to reproduce this?

**Comment by sppalkia** [comment]
Hi Mihai,

We are able to reproduce the issue - we're looking into a patch for it right now. The issue seems to occur with vectorization on parallel code with a fringe (the % 4 issue you pointed out), and only seems to occur with the appender. Thanks for pointing it out!

**Comment by mihai-varga** [comment]
Thanks for looking into it!

**Comment by sppalkia** [comment]
Resolved by #318
